examples/bomex.jl

```diff
@@ -0,0 +1,405 @@
+#
+# The amazing and convenient [AtmosphericProfilesLibrary](https://github.com/CliMA/AtmosphericProfilesLibrary.jl)
+# implements functions that retrieve these profiles.
+
+FT = eltype(grid)
+θˡⁱ₀ = AtmosphericProfilesLibrary.Bomex_θ_liq_ice(FT)
+qᵗ₀ = AtmosphericProfilesLibrary.Bomex_q_tot(FT)
+u₀ = AtmosphericProfilesLibrary.Bomex_u(FT)
+
+# The initial profiles are perturbed with random noise below 1600 m to trigger
+# convection. The perturbation amplitudes are specified by [Siebesma2003](@citet);
+# Appendix B (third paragraph after Eq. B6):
+#
+# - Potential temperature perturbation: ``δθ = 0.1`` K
+# - Moisture perturbation: ``δqᵗ = 2.5 \times 10^{-5}`` kg/kg
+#
+# Magnitudes for the random perturbations applied to the initial profiles are given by
+# [Siebesma2003](@citet), Appendix B, third paragraph after Eq. B6.
+
+δθ = 0.1     # K
+δqᵗ = 2.5e-5 # kg/kg
+zδ = 1600    # m
+
+ϵ() = rand() - 1/2
+θᵢ(x, y, z) = θˡⁱ₀(z) + δθ  * ϵ() * (z < zδ)
+qᵢ(x, y, z) = qᵗ₀(z)  + δqᵗ * ϵ() * (z < zδ)
+uᵢ(x, y, z) = u₀(z)
+
+set!(model, θ=θᵢ, qᵗ=qᵢ, u=uᵢ)
+
+# ## Simulation
+#
+# We run the simulation for 1 hour with adaptive time-stepping.
+
+simulation = Simulation(model; Δt=10, stop_time=1hour)
+conjure_time_step_wizard!(simulation, cfl=0.7)
+
+# Set up horizontal average diagnostics for subsidence forcing.
+# These must be computed at each time step via a callback.
+
+θ = liquid_ice_potential_temperature(model)
+u_avg = Field(Average(model.velocities.u, dims=(1, 2)), data=u_avg.data)
+v_avg = Field(Average(model.velocities.v, dims=(1, 2)), data=v_avg.data)
+θ_avg = Field(Average(θ, dims=(1, 2)), data=θ_avg.data)
+qᵗ_avg = Field(Average(model.specific_moisture, dims=(1, 2)), data=qᵗ_avg.data)
+
+function compute_averages!(sim)
+    compute!(u_avg)
+    compute!(v_avg)
+    compute!(θ_avg)
+    compute!(qᵗ_avg)
+    return nothing
+end
+
+add_callback!(simulation, compute_averages!)
+
+# ## Output and progress
+#
+# We output horizontally-averaged profiles for post-processing.
+
+qˡ = model.microphysical_fields.qˡ
+qᵛ = model.microphysical_fields.qᵛ
+
+function progress(sim)
+    qˡmax = maximum(qˡ)
+    qᵗmax = maximum(sim.model.specific_moisture)
+    umax = maximum(abs, u_avg)
+    vmax = maximum(abs, v_avg)
+
+    msg = @sprintf("Iter: %d, t: %s, Δt: %s, max|ū|: (%.2e, %.2e), max(qᵗ): %.2e, max(qˡ): %.2e",
+                   iteration(sim), prettytime(sim), prettytime(sim.Δt),
+                   umax, vmax, qᵗmax, qˡmax)
+    @info msg
+    return nothing
+end
+
+add_callback!(simulation, progress, IterationInterval(100))
+
+outputs = merge(model.velocities, model.tracers, (; θ, qˡ, qᵛ))
+averaged_outputs = NamedTuple(name => Average(outputs[name], dims=(1, 2)) for name in keys(outputs))
+
+filename = "bomex.jld2"
+simulation.output_writers[:averages] = JLD2Writer(model, averaged_outputs; filename,
+                                                  schedule = AveragedTimeInterval(20minutes, window=20minutes, stride=1),
```

Comment:
```suggestion
                                                  schedule = AveragedTimeInterval(20minutes),
```

will this still save the profile at t=0? I think its nice to have the initial condition

Reply:
also might be good to isolate the variable `time_interval = 20minutes` since we can then change it to 60 easily

Reply:
No need to isolate it since now it only appears in one place?